unlock-app/src/components/interface/keychain/KeyMetadataDrawer.tsx

```diff
@@ -0,0 +1,331 @@
+import { Transition, Dialog } from '@headlessui/react'
+import { RiCloseLine as CloseIcon } from 'react-icons/ri'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+
+interface MetadataProps {
+  tokenId: string
+  network: number
+  lock: any
+  account: string
+}
+
+interface MetadataDrawerProps {
+  isOpen: boolean
+  setIsOpen: (open: boolean) => void
+  tokenId: string
+  network: number
+  lock: any
+  account: string
+}
+
+const KeyMetadataPlaceholder: React.FC<unknown> = () => {
+  return (
+    <div className="flex flex-col gap-3 p-4">
+      <div className="w-[120px] h-[24px] bg-slate-200 animate-pulse"></div>
+      <div className="w-full h-[30px] rounded-lg bg-slate-200 animate-pulse"></div>
+      <div className="w-full h-[30px] rounded-lg bg-slate-200 animate-pulse"></div>
+      <div className="w-full h-[30px] rounded-lg bg-slate-200 animate-pulse"></div>
+      <div className="w-full h-[30px] rounded-lg bg-slate-200 animate-pulse"></div>
+      <div className="flex">
+        <div className="ml-auto w-[100px] h-[40px] bg-slate-200 animate-pulse rounded-full"></div>
+      </div>
+    </div>
+  )
+}
+
+interface MetadataPropertiesProps {
+  lockAddress: string
+  network: number
+  isLoading: boolean
+}
+
+const Header = ({ title }: any) => {
+  return (
+    <div className="flex flex-col">
+      <span className="text-xl font-bold text-brand-ui-primary">{title}</span>
+    </div>
+  )
+}
+
+const Link = ({ url, label }: any) => {
+  return (
+    <a
+      className="text- text-brand-ui-primary"
+      href={url}
+      target="_blank"
+      rel="noreferrer"
+    >
+      {label}
+    </a>
+  )
+}
+
+const PublicLockProperties = ({
+  lockAddress,
+  network,
+  isLoading,
+}: MetadataPropertiesProps) => {
+  const storageService = useStorageService()
+  const { data } = useQuery<Record<string, any>>(
+    ['lockMetadata', lockAddress, network],
+    async () => {
+      const response = await storageService.locksmith.lockMetadata(
+        network,
+        lockAddress
+      )
+      return response.data
+    },
+    {
+      onError() {
+        ToastHelper.error('Impossible to retrieve public NFT metadata.')
+      },
+      refetchInterval: Infinity,
+      refetchOnWindowFocus: false,
+      refetchOnReconnect: false,
+      retry: 2,
+    }
+  )
+
+  const hasLinks =
+    data?.external_url || data?.youtube_url || data?.animation_url
+
+  const attributes: {
+    value: string
+    trait_type: string
+    max_value?: number
+    display_type?: string
+  }[] = data?.attributes ?? []
+
+  const propertyAttributes = attributes.filter(
+    (item) => item.trait_type && item.value && !item?.max_value
+  )
+
+  const levelsAttributes = attributes?.filter(
+    (item) =>
+      item.trait_type && item.value && item.max_value && !item.display_type
+  )
+
+  const statsAttributes = attributes?.filter(
+    (item) =>
+      item.trait_type && item.value && item.max_value && item?.display_type
+  )
```

Comment:
You can use existing function from the utils for metadata form for this.

Reply:
changed! thanks!